### PR TITLE
Improve phpdoc for ClassMetadataInfo

### PR DESCRIPTION
**Pull request by VincentLanglet**
Hi @greg0ire, this is a follow up of https://github.com/doctrine/orm/pull/9794.

Cc @derrabus https://github.com/doctrine/orm/pull/9794#issuecomment-1141505104 There are less optional key.

Static analysis build is green. Tests are failing but this can't be related to my changes since I only touch phpdodc

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 1]
```diff
@@ -99,36 +99,36 @@
  * }
  * @psalm-type AssociationMapping = array{
  *     cache?: array,
- *     cascade?: array<string>,
+ *     cascade: array<string>,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:112]: set in _validateAndCompleteAssociationMapping

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 2]
```diff
@@ -99,36 +99,36 @@
...
  *     declared?: class-string,
- *     fetch?: mixed,
+ *     fetch: mixed,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:114]: set in _validateAndCompleteAssociationMapping

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 3]
```diff
@@ -99,36 +99,36 @@
...
  *     id?: bool,
  *     inherited?: class-string,
  *     indexBy?: string,
- *     inversedBy?: string|null,
- *     isCascadeRemove?: bool,
- *     isCascadePersist?: bool,
- *     isCascadeRefresh?: bool,
- *     isCascadeMerge?: bool,
- *     isCascadeDetach?: bool,
+ *     inversedBy: string|null,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:119]: null is set by default in _validateAndCompleteAssociationMapping

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 4]
```diff
@@ -99,36 +99,36 @@
...
+ *     isCascadeRemove: bool,
+ *     isCascadePersist: bool,
+ *     isCascadeRefresh: bool,
+ *     isCascadeMerge: bool,
+ *     isCascadeDetach: bool,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:124]: set in _validateAndCompleteAssociationMapping

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 5]
```diff
@@ -99,36 +99,36 @@
...
  *     isOnDeleteCascade?: bool,
- *     isOwningSide?: bool,
+ *     isOwningSide: bool,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:126]: set by default to true in _validateAndCompleteAssociationMapping (and then changed to false if needed)

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 6]
```diff
@@ -99,36 +99,36 @@
...
  *     joinColumns?: array,
- *     joinColumnFieldNames?: array,
+ *     joinColumnFieldNames?: array<string, string>,
  *     joinTable?: array,
  *     joinTableColumns?: list<mixed>,
- *     mappedBy?: string|null,
+ *     mappedBy: string|null,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:131]: null is set by default in _validateAndCompleteAssociationMapping

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 7]
```diff
@@ -99,36 +99,36 @@
...
  *     orderBy?: array,
  *     originalClass?: class-string,
  *     originalField?: string,
  *     orphanRemoval?: bool,
  *     relationToSourceKeyColumns?: array,
  *     relationToTargetKeyColumns?: array,
- *     sourceEntity?: class-string,
- *     sourceToTargetKeyColumns?: array,
+ *     sourceEntity: class-string,
```
Comment (VincentLanglet) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:138]: set in _validateAndCompleteAssociationMapping

**Review by derrabus — DISMISSED** [review]

**Comment by greg0ire** [comment]
Hi @VincentLanglet I'll be AFK for a week, I saw there is a PR to fix the build, we can merge yours after that one

**Comment by greg0ire** [comment]
See https://github.com/doctrine/orm/pull/9966

**Comment by VincentLanglet** [comment]
There is a lot of new errors, even with static analysis, but none are related to the PR.
It certainly related to some new released since some Deprecated class/method are reported.

I'll wait for
>  See https://github.com/doctrine/orm/pull/9966

then

**Comment by derrabus** [comment]
I took the liberty to rebase and squash your PR after having fixed the build on the 2.13.x branch. If the build is green, I'll merge.

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you @VincentLanglet!

**Pull request merged**